pkg/builder/csolution/builder.go

```diff
@@ -366,12 +364,14 @@ func (b CSolutionBuilder) Build() (err error) {
 		return err
 	}
 
-	// get list of selected contexts
-	idxFile, err := b.getIdxFilePath()
+	projName, err := utils.GetProjectName(b.InputFile)
 	if err != nil {
 		return err
 	}
-	selectedContexts, err := b.getSelectedContexts(idxFile)
+	setFile := utils.NormalizePath(filepath.Join(filepath.Dir(b.InputFile), projName+".cbuild-set.yml"))
+
+	// get list of selected contexts
+	selectedContexts, err := b.getSelectedContexts(setFile)
```

Comment:
After the introduction of https://github.com/Open-CMSIS-Pack/cbuild/pull/103 the `*.cbuild-set.yml` file should be read conditionally only when the `-S, --context-set` flag is set.